### PR TITLE
on premises automation has been removed by default

### PR DESCRIPTION
**Pull request by testisnullus**

### CODE REVIEW

File: apis/clusters/v1beta1/cadence_webhook.go [hunk 1]
```diff
@@ -86,7 +89,16 @@ func (cv *cadenceValidator) ValidateCreate(ctx context.Context, obj runtime.Obje
 		return err
 	}
 
-	if c.Spec.OnPremisesSpec != nil {
+	exists, err := KubeVirtAddonExists(ctx, cv.Client)
```
Comment (ribaraka) [inline apis/clusters/v1beta1/cadence_webhook.go:92]: ContainsKubeVirtAddon

File: apis/clusters/v1beta1/kafka_webhook.go [hunk 2]
```diff
@@ -85,7 +87,16 @@ func (kv *kafkaValidator) ValidateCreate(ctx context.Context, obj runtime.Object
...
 
-	if k.Spec.OnPremisesSpec != nil {
+	exists, err := KubeVirtAddonExists(ctx, kv.Client)
+	if err != nil {
+		return err
+	}
+
+	if k.Spec.OnPremisesSpec != nil && k.Spec.OnPremisesSpec.EnableAutomation {
+		if !exists {
+			return fmt.Errorf("cannot create KubeVirt based resources automatially without KubeVirt operator " +
```
Comment (ribaraka) [inline apis/clusters/v1beta1/kafka_webhook.go:97]: make a var of it

File: apis/clusters/v1beta1/validation.go [hunk 3]
```diff
@@ -253,3 +265,41 @@ func (dc *DataCentre) ValidateOnPremisesCreation() error {
...
+	}
+
+	deployments := &k8sappsv1.DeploymentList{}
+	err = client.List(ctx, deployments)
+	if err != nil {
+		return false, err
+	}
+
+	for _, deployment := range deployments.Items {
+		if checkKubeVirtLabels(deployment.Labels) || checkKubeVirtLabels(deployment.Annotations) {
```
Comment (ribaraka) [inline apis/clusters/v1beta1/validation.go:289]: containsKubeVirtLabels

File: controllers/clusters/kafka_controller.go [hunk 4]
```diff
@@ -519,9 +519,9 @@ func (r *KafkaReconciler) handleDeleteCluster(ctx context.Context, k *v1beta1.Ka
 		return reconcile.Result{}, err
 	}
 
-	if k.Spec.OnPremisesSpec != nil {
+	if k.Spec.OnPremisesSpec != nil && k.Spec.OnPremisesSpec.EnableAutomation {
 		err = deleteOnPremResources(ctx, r.Client, k.Status.ID, k.Namespace)
-		if err != nil {
+		if client.IgnoreNotFound(err) != nil {
```
Comment (ribaraka) [inline controllers/clusters/kafka_controller.go:524]: revert it pls

File: controllers/clusters/cadence_controller.go [hunk 5]
```diff
@@ -549,7 +549,7 @@ func (r *CadenceReconciler) handleDeleteCluster(
 				"Two-Factor Delete is enabled, please confirm cluster deletion via email or phone.")
 			return ctrl.Result{}, nil
 		}
-		if c.Spec.OnPremisesSpec != nil {
+		if c.Spec.OnPremisesSpec != nil && c.Spec.OnPremisesSpec.EnableAutomation {
```
Comment (ribaraka) [inline controllers/clusters/cadence_controller.go:552]: add the same condition for creation

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/cadence_webhook.go (hunk 1 above)
Comment (testisnullus) [reply apis/clusters/v1beta1/cadence_webhook.go]: Agreed, fixed

File: apis/clusters/v1beta1/kafka_webhook.go (hunk 2 above)
Comment (testisnullus) [reply apis/clusters/v1beta1/kafka_webhook.go]: fixed

File: apis/clusters/v1beta1/validation.go (hunk 3 above)
Comment (testisnullus) [reply apis/clusters/v1beta1/validation.go]: ok, fixed

File: controllers/clusters/kafka_controller.go (hunk 4 above)
Comment (testisnullus) [reply controllers/clusters/kafka_controller.go]: Ok, fixed

File: controllers/clusters/cadence_controller.go (hunk 5 above)
Comment (testisnullus) [reply controllers/clusters/cadence_controller.go:552]: Added

**Review by ribaraka — APPROVED** [review]

**Review by taaraora — APPROVED** [review]

**Pull request merged**